financialaid/views.py

```diff
@@ -1,24 +1,229 @@
+
+    def get_context_data(self, **kwargs):
+        """
+        Gets context for view
+        """
+        context = super().get_context_data(**kwargs)
+
+        # Constants required in view
+        context["selected_status"] = self.selected_status
+        context["current_program_id"] = self.program.id
+
+        # Create ordered list of (financial aid status, financial message)
+        messages = FinancialAidStatus.STATUS_MESSAGES_DICT
+        message_order = (
+            FinancialAidStatus.PENDING_MANUAL_APPROVAL,
+            FinancialAidStatus.APPROVED,
+            FinancialAidStatus.REJECTED,
+            FinancialAidStatus.AUTO_APPROVED
+        )
+        context["financial_aid_statuses"] = (
+            (status, "Show: {message}".format(message=messages[status]))
+            for status in message_order
+        )
+
+        # Get sort field information
+        sort_link_order = (
+            "first_name",
+            "last_name",
+            "location",
+            "adjusted_cost",
+            "reported_income",
+        )
+        new_sort_direction = "" if self.sort_direction == "-" else "-"
+        context["sort_field_info"] = (
+            {
+                # For appending the sort_by get param on url
+                "sort_field": "{sort_direction}{sort_field}".format(
+                    # If this field is our current sort field, we want to toggle the sort direction, else default ""
+                    sort_direction=new_sort_direction if field == self.sort_field else "",
+                    sort_field=field
+                ),
+                # If this field is the current sort field, we want to indicate the current sort direction
+                "direction_display": self.sort_direction if field == self.sort_field else None,
+                "display": self.sort_fields[field]["display"]
+            }
+            for field in sort_link_order
+        )
+
+        # Required for styling
+        context["style_src"] = get_bundle_url(self.request, "style.js")
+        context["dashboard_src"] = get_bundle_url(self.request, "dashboard.js")
+        js_settings = {
+            "gaTrackingID": settings.GA_TRACKING_ID,
+            "reactGaDebug": settings.REACT_GA_DEBUG,
+            "authenticated": not self.request.user.is_anonymous(),
+            "edx_base_url": settings.EDXORG_BASE_URL,
+        }
+        context["js_settings_json"] = json.dumps(js_settings)
+        return context
+
+    def get_queryset(self):
+        """
+        Gets queryset for ListView to return to view
+        """
+        # Filter by availability
+        financial_aids = FinancialAid.objects.filter(
+            tier_program__program__live=True,
+            tier_program__program__financial_aid_availability=True
+        )
+
+        # Filter by program (self.program set in test_func())
+        financial_aids = financial_aids.filter(
```

Comment:
Ah - I had them separate for readibility, but I forgot about the potential issues with chaining filters on foreign keys.  I'll combine these.